instrumentation/iceberg-1.8/library/src/main/java/io/opentelemetry/instrumentation/iceberg/v1_8/ScanMetricsBuilder.java

```diff
@@ -0,0 +1,174 @@
+  private static final String TOTAL_DELETE_FILE_SIZE_IN_BYTES = ROOT + ".scanned.delete_files.size";
+  private static final String SKIPPED_DATA_MANIFESTS = ROOT + ".skipped.data_manifests.count";
+  private static final String SKIPPED_DELETE_MANIFESTS = ROOT + ".skipped.delete_manifests.count";
+  private static final String SKIPPED_DATA_FILES = ROOT + ".skipped.data_files.count";
+  private static final String SKIPPED_DELETE_FILES = ROOT + ".skipped.delete_files.count";
+  private static final String INDEXED_DELETE_FILES = ROOT + ".scanned.indexed_delete_files.count";
+  private static final String EQUALITY_DELETE_FILES = ROOT + ".scanned.equality_delete_files.count";
+  private static final String POSITIONAL_DELETE_FILES =
+      ROOT + ".scanned.positional_delete_files.count";
+  private static final String DVS = ROOT + ".scanned.dvs.count";
+
+  private ScanMetricsBuilder() {
+    // prevents instantiation
+  }
+
+  static LongGauge totalPlanningDuration(Meter meter, String unit) {
+    return meter
+        .gaugeBuilder(TOTAL_PLANNING_DURATION)
+        .setDescription("The total duration needed to plan the scan.")
+        .setUnit(unit)
+        .ofLongs()
+        .build();
+  }
+
+  static LongCounter scannedDataFilesCount(Meter meter) {
+    return meter
+        .counterBuilder(RESULT_DATA_FILES)
+        .setDescription("The number of scanned data files.")
+        .setUnit("{file}")
+        .build();
+  }
+
+  static LongCounter scannedDeleteFilesCount(Meter meter) {
+    return meter
+        .counterBuilder(RESULT_DELETE_FILES)
+        .setDescription("The number of scanned delete files.")
+        .setUnit("{file}")
+        .build();
+  }
+
+  static LongCounter scannedDataManifestsCount(Meter meter) {
+    return meter
+        .counterBuilder(SCANNED_DATA_MANIFESTS)
+        .setDescription("The number of scanned data manifests.")
+        .setUnit("{file}")
+        .build();
+  }
+
+  static LongCounter scannedDeleteManifestsCount(Meter meter) {
+    return meter
+        .counterBuilder(SCANNED_DELETE_MANIFESTS)
+        .setDescription("The number of scanned delete manifests.")
+        .setUnit("{file}")
+        .build();
+  }
+
+  static LongCounter totalDataManifestsCount(Meter meter) {
+    return meter
+        .counterBuilder(TOTAL_DATA_MANIFESTS)
+        .setDescription("The number of all data manifests.")
+        .setUnit("{file}")
+        .build();
+  }
+
+  static LongCounter totalDeleteManifestsCount(Meter meter) {
+    return meter
+        .counterBuilder(TOTAL_DELETE_MANIFESTS)
+        .setDescription("The number of all delete manifests.")
+        .setUnit("{file}")
+        .build();
+  }
+
+  static LongCounter scannedDataFilesSize(Meter meter) {
+    return meter
+        .counterBuilder(TOTAL_FILE_SIZE_IN_BYTES)
+        .setDescription("The total size of all scanned data files.")
+        .setUnit("byte")
```

Comment:
bytes are typically represented by "By"

```suggestion
        .setUnit("By")
```